### PR TITLE
Feature/228 more flexible pnn

### LINKED ISSUE #228: Enable PNN without weather information
**What is your feature request related to?**
* Module request

**Please describe the module/problem**
I would like to use the PNN without weather information. Perhaps, the input can be made flexible.

### PR DESCRIPTION
**Pull request by benHeid**
## Related Issues
closes #228 
## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos)
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Did you update the CHANGELOG

## PR review
Anyone in the community is free to review the PR once the tests have passed.

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones to the PR so it can be classified

### CODE REVIEW

File: pywatts/modules/models/profile_neural_network.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def set_params(self, epochs=None, offset=None, batch_size=None, validation_split
         if validation_split:
             self.validation_split = validation_split
 
-    def transform(self, historical_input, calendar, temperature, humidity, profile, trend) -> xr.DataArray:
+    def transform(self, **kwargs) -> xr.DataArray:
         """
         Forecast the electrical load for the given input.
```
Comment (marianturowski) [inline pywatts/modules/models/profile_neural_network.py:82]: Do you want to leave the list of the inputs (in lines 84 to 95) as it is or do you want to change it kwargs accordingly?

File: pywatts/modules/models/profile_neural_network.py [hunk 2]
```diff
@@ -95,17 +95,19 @@ def transform(self, historical_input, calendar, temperature, humidity, profile,
...
+            "profile": profile,
+            "dummy_input": dummy_input
         })
-        return numpy_to_xarray(result, historical_input)
+        return numpy_to_xarray(result, reference)
 
-    def fit(self, historical_input, calendar, temperature, humidity, profile, trend, target):
+    def fit(self, target, **kwargs):
         """
         Fit the Profile Neural Network.
```
Comment (marianturowski) [inline pywatts/modules/models/profile_neural_network.py:113]: Lines 114 to 125: see above? Adapt it to kwargs?

**Review by marianturowski — CHANGES_REQUESTED** [review]
Thank you for making the PNN more flexible.
Please consider adapting the comments in the code as I suggested.

File: pywatts/modules/models/profile_neural_network.py [hunk 3]
```diff
@@ -124,22 +126,35 @@ def fit(self, historical_input, calendar, temperature, humidity, profile, trend,
...
-                 humidity.values.reshape(-1, self.horizon, 1)], axis=-1)[self.offset:]
+            "hist_input": historical_input,
+            "full_trend": trend,
+            "profile": profile,
+            "dummy_input": dummy_input
         }, target.values[self.offset:])
         self.pnn.fit(input, t, epochs=self.epochs, batch_size=self.batch_size, validation_split=self.validation_split)
         self.is_fitted = True
 
+    def _get_inputs(self, kwargs, offset):
```
[the remark this answers is not in the document]
Comment (benHeid) [reply pywatts/modules/models/profile_neural_network.py:147]: I added a comment

File: pywatts/modules/models/profile_neural_network.py (hunk 2 above)
Comment (marianturowski) [reply pywatts/modules/models/profile_neural_network.py:115]: I see your point. That's fine.

File: pywatts/modules/models/profile_neural_network.py (hunk 1 above)
Comment (marianturowski) [reply pywatts/modules/models/profile_neural_network.py:83]: I see your point. That's fine.

File: pywatts/modules/models/profile_neural_network.py [hunk 4]
```diff
@@ -78,7 +78,8 @@ def set_params(self, epochs=None, offset=None, batch_size=None, validation_split
 
     def transform(self, **kwargs) -> xr.DataArray:
         """
-        Forecast the electrical load for the given input.
+        Forecast the electrical load for the given input. Note historical_input, trend, and profile has to be provided.
```
Comment (marianturowski) [inline pywatts/modules/models/profile_neural_network.py:81]: Please change:
Forecast the electrical load for the given input. Note that historical_input, trend, and profile have to be provided.

File: pywatts/modules/models/profile_neural_network.py [hunk 5]
```diff
@@ -109,7 +110,8 @@ def transform(self, **kwargs) -> xr.DataArray:
 
     def fit(self, target, **kwargs):
         """
-        Fit the Profile Neural Network.
+        Fit the Profile Neural Network. Note historical_input, trend, and profile has to be provided.
```
Comment (marianturowski) [inline pywatts/modules/models/profile_neural_network.py:113]: Fit the Profile Neural Network. Note that historical_input, trend, and profile have to be provided.

**Review by marianturowski — CHANGES_REQUESTED** [review]
I suggest two small changes in new comments

**Pull request merged**